geocity/views.py

```diff
@@ -86,7 +86,55 @@ def get_context_data(self, **kwargs):
         return context
 
 
+def get_redirect_url_after_login(qs_dict):
+    url_value = (
+        qs_dict["next"][0]
+        if "next" in qs_dict
+        else reverse("permits:permit_request_select_administrative_entity")
+    )
+    return url_value
+
+
+def create_duo_client(username, request):
+    # Initial value of duo_client
+    duo_client = None
+
+    # Get user corresponding to the user name
+    user = User.objects.get(username=username)
+
+    # Get base queryset
+    permitdepartement_qs = models.PermitDepartment.objects.filter(
+        group__in=user.groups.all(), mandatory_2fa=True, duo_config__isnull=False
+    )
+
+    if permitdepartement_qs.exists():
+        # Get PermitDepartments using an active duo configuration. Used to retrieve data for duo_universal.Client()
+        permitdepartement = permitdepartement_qs.filter(duo_config__is_active=True)
+        duo_mfa = permitdepartement.exists()
+    else:
+        # Tells if duo multi factor authentication is activated
+        duo_mfa = False
+
+    if duo_mfa:
+        # Retrieve information for duo_universal.Client from first PermitDepartment
+        client_id = permitdepartement[0].duo_config.client_id
+        client_secret = permitdepartement[0].duo_config.client_secret
+        host = permitdepartement[0].duo_config.host
+
+        # Define the callback
+        callback_url = request.build_absolute_uri(reverse("duo_callback"))
+
+        # Create the duo_universal.Client
+        duo_client = duo_universal.Client(client_id, client_secret, host, callback_url)
+    return user, duo_mfa, duo_client
```

Comment:
@danickfort could you please double check that part ?

Reply:
This looks OK code-wise, but I can't test it right now.